packages/contracts/contracts/YieldGate.sol

```diff
@@ -93,23 +109,62 @@ contract BeneficiaryPool {
 
         gate = YieldGate(_gate);
         beneficiary = _beneficiary;
+
+        emitParametersChanged(0, 0);
+    }
+
+    // To save gas, add individual parameter setters.
+
+    function setMinAmount(uint256 _minAmount) external onlyBeneficiary {
+        minAmount = _minAmount;
+        emitParametersChanged(_minAmount, minDuration);
+    }
+
+    function setMinDuration(uint256 _minDuration) external onlyBeneficiary {
+        minDuration = _minDuration;
+        emitParametersChanged(minAmount, _minDuration);
+    }
+
+    function setParameters(uint256 _minAmount, uint256 _minDuration) external onlyBeneficiary {
+        minAmount = _minAmount;
+        minDuration = _minDuration;
+        emitParametersChanged(_minAmount, _minDuration);
+    }
+
+    function emitParametersChanged(uint256 _minAmount, uint256 _minDuration) internal {
+        emit ParametersChanged(beneficiary, _minAmount, _minDuration);
     }
 
     // Stakes the sent ether on behalf of the provided supporter. The supporter
     // is usually msg.sender if staking on the transaction sender's behalf.
+    // The staking timeout is reset on each call, so prior stake is re-locked.
     function stake(address supporter) public payable {
         uint256 amount = msg.value;
+        require(amount > 0 && amount >= minAmount, "amount too low");
+
         supporters[supporter] += amount;
         totalStake += amount;
+        uint256 timeout = 0;
+        if (minDuration > 0) {
+            timeout = block.timestamp + minDuration;
```

Comment:
This also always resets the timeout when somebody tops-up, am I right? Which is kind of mean, isn't it 😆

Reply:
Yes it does reset on each top up, also added this fact to the function comment. I left it like that on purpose, because the alternative is to do more complex accounting of each top up, including storing each individual top up with lockTimeouts in an array etc... would get super messy and I want to keep things simple in this iteration.

Also, IMO this is actually a rather unimportant case to consider. I mean, if you top up, it seems like you want to support more, so why would you insist on getting the old stake out earlier? If that's the case, then better not top up at all and get your old stake out first ^^ I think in for first iteration it is sufficient to give the user a notice in the frontend that in case of a top up, the locked timeout is reset for their whole stake.

Reply:
Agreed